pkg/resultstore/fs.go

```diff
@@ -50,3 +50,12 @@ func (fs osFS) OpenRead(name string) (io.ReadCloser, error) {
 func (fs osFS) ListDirEntries(name string) ([]fs.DirEntry, error) {
 	return os.ReadDir(name)
 }
+
+func (fs osFS) openFileMkdirAll(name string, flags int, perm fs.FileMode) (io.WriteCloser, error) {
+	path := filepath.Join(fs.dir, name)
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		dir, _ := filepath.Split(path)
```

Comment:
Recommend filepath.Dir instead of Split as you're only interested in the directory.

Barely any performance difference. It's only for the readability of not having a write-only variable that could be confused with ignoring an error.

```suggestion
		dir := filepath.Dir(path)
```